server/core/src/ldaps.rs

```diff
@@ -112,28 +118,65 @@ async fn client_process<STREAM>(
 }
 
 async fn client_tls_accept(
-    tcpstream: TcpStream,
+    stream: TcpStream,
     tls_acceptor: SslAcceptor,
-    client_socket_addr: net::SocketAddr,
+    connection_addr: SocketAddr,
     qe_r_ref: &'static QueryServerReadV1,
+    trusted_haproxy_ips: Option<Arc<HashSet<IpAddr>>>,
```

Comment:
See similar changes above. It's not a haproxy header/ip list, it's the proxy protocol.